### ISSUE TITLE
Chrome "Invalid mapping of id/email" in tight loop

### ISSUE THREAD

**Issue by lambdaupb** [microg/android_packages_apps_GmsCore#76]
My logcat is flooded with messages like this (below).
Interestingly I have explicitly disabled the use of my account in chrome, so there should be no reason for chrome to request any token.

I really have no clue at the moment what could cause this.

But a blacklist feature based on `clientPackageName` would make it possible to ignore misbehaving apps.

```
01-13 15:17:35.574 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.576 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.580 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.582 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.583 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.585 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.588 D/GmsAuthManagerSvc( 2727): getToken: account:myaccountname@gmail.com scope:^^_account_id_^^ extras:Bundle[{androidPackageName=com.android.chrome, clientPackageName=com.android.chrome, service_connection_start_time_millis=912270822}], notify: false
01-13 15:17:35.592 D/GmsAuthManagerSvc( 2727): getToken: INVALID_TOKEN
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): AccountIdProvider.getAccountId
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): com.google.android.gms.auth.a: OK
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): 	at com.google.android.gms.auth.b.a(Unknown Source)
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): 	at com.google.android.gms.auth.b.a(Unknown Source)
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountIdProvider.getAccountId(AccountIdProvider.java:47)
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountTrackerService$1.doInBackground(AccountTrackerService.java:110)
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountTrackerService$1.doInBackground(AccountTrackerService.java:104)
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): 	at android.os.AsyncTask$2.call(AsyncTask.java:292)
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
01-13 15:17:35.594 E/cr_AccountIdProvider(17799): 	at java.lang.Thread.run(Thread.java:818)
01-13 15:17:35.594 W/cr_AccountService(17799): Invalid mapping of id/email
01-13 15:17:35.595 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.596 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.600 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.602 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.603 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.605 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.607 D/GmsAuthManagerSvc( 2727): getToken: account:myaccountname@gmail.com scope:^^_account_id_^^ extras:Bundle[{androidPackageName=com.android.chrome, clientPackageName=com.android.chrome, service_connection_start_time_millis=912270841}], notify: false
01-13 15:17:35.611 D/GmsAuthManagerSvc( 2727): getToken: INVALID_TOKEN
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): AccountIdProvider.getAccountId
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): com.google.android.gms.auth.a: OK
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): 	at com.google.android.gms.auth.b.a(Unknown Source)
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): 	at com.google.android.gms.auth.b.a(Unknown Source)
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountIdProvider.getAccountId(AccountIdProvider.java:47)
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountTrackerService$1.doInBackground(AccountTrackerService.java:110)
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountTrackerService$1.doInBackground(AccountTrackerService.java:104)
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): 	at android.os.AsyncTask$2.call(AsyncTask.java:292)
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
01-13 15:17:35.613 E/cr_AccountIdProvider(17799): 	at java.lang.Thread.run(Thread.java:818)
01-13 15:17:35.613 W/cr_AccountService(17799): Invalid mapping of id/email
01-13 15:17:35.614 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.616 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.620 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.622 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.623 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.625 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.627 D/GmsAuthManagerSvc( 2727): getToken: account:myaccountname@gmail.com scope:^^_account_id_^^ extras:Bundle[{androidPackageName=com.android.chrome, clientPackageName=com.android.chrome, service_connection_start_time_millis=912270861}], notify: false
01-13 15:17:35.631 D/GmsAuthManagerSvc( 2727): getToken: INVALID_TOKEN
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): AccountIdProvider.getAccountId
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): com.google.android.gms.auth.a: OK
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): 	at com.google.android.gms.auth.b.a(Unknown Source)
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): 	at com.google.android.gms.auth.b.a(Unknown Source)
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountIdProvider.getAccountId(AccountIdProvider.java:47)
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountTrackerService$1.doInBackground(AccountTrackerService.java:110)
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountTrackerService$1.doInBackground(AccountTrackerService.java:104)
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): 	at android.os.AsyncTask$2.call(AsyncTask.java:292)
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
01-13 15:17:35.634 E/cr_AccountIdProvider(17799): 	at java.lang.Thread.run(Thread.java:818)
01-13 15:17:35.634 W/cr_AccountService(17799): Invalid mapping of id/email
01-13 15:17:35.635 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.636 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.640 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.641 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.642 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.644 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.646 D/GmsAuthManagerSvc( 2727): getToken: account:myaccountname@gmail.com scope:^^_account_id_^^ extras:Bundle[{androidPackageName=com.android.chrome, clientPackageName=com.android.chrome, service_connection_start_time_millis=912270881}], notify: false
01-13 15:17:35.651 D/GmsAuthManagerSvc( 2727): getToken: INVALID_TOKEN
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): AccountIdProvider.getAccountId
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): com.google.android.gms.auth.a: OK
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): 	at com.google.android.gms.auth.b.a(Unknown Source)
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): 	at com.google.android.gms.auth.b.a(Unknown Source)
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountIdProvider.getAccountId(AccountIdProvider.java:47)
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountTrackerService$1.doInBackground(AccountTrackerService.java:110)
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountTrackerService$1.doInBackground(AccountTrackerService.java:104)
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): 	at android.os.AsyncTask$2.call(AsyncTask.java:292)
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
01-13 15:17:35.653 E/cr_AccountIdProvider(17799): 	at java.lang.Thread.run(Thread.java:818)
01-13 15:17:35.653 W/cr_AccountService(17799): Invalid mapping of id/email
01-13 15:17:35.654 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.655 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.659 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.660 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.662 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.663 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.666 D/GmsAuthManagerSvc( 2727): getToken: account:myaccountname@gmail.com scope:^^_account_id_^^ extras:Bundle[{androidPackageName=com.android.chrome, clientPackageName=com.android.chrome, service_connection_start_time_millis=912270900}], notify: false
01-13 15:17:35.670 D/GmsAuthManagerSvc( 2727): getToken: INVALID_TOKEN
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): AccountIdProvider.getAccountId
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): com.google.android.gms.auth.a: OK
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): 	at com.google.android.gms.auth.b.a(Unknown Source)
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): 	at com.google.android.gms.auth.b.a(Unknown Source)
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountIdProvider.getAccountId(AccountIdProvider.java:47)
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountTrackerService$1.doInBackground(AccountTrackerService.java:110)
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): 	at org.chromium.chrome.browser.signin.AccountTrackerService$1.doInBackground(AccountTrackerService.java:104)
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): 	at android.os.AsyncTask$2.call(AsyncTask.java:292)
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
01-13 15:17:35.672 E/cr_AccountIdProvider(17799): 	at java.lang.Thread.run(Thread.java:818)
01-13 15:17:35.672 W/cr_AccountService(17799): Invalid mapping of id/email
01-13 15:17:35.673 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.675 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.678 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.680 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.681 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.683 I/Xposed  ( 2185): FakeGApps: faked signature
01-13 15:17:35.685 D/GmsAuthManagerSvc( 2727): getToken: account:myaccountname@gmail.com scope:^^_account_id_^^ extras:Bundle[{androidPackageName=com.android.chrome, clientPackageName=com.android.chrome, service_connection_start_time_millis=912270920}], notify: false
01-13 15:17:35.690 D/GmsAuthManagerSvc( 2727): getToken: INVALID_TOKEN
```

**Comment by lambdaupb** [comment]
So I did some research, it seems Chrome indirectly uses the GoogleAuthUtil class and the finally this piece of code in an internal class of the play-services:
```
    public static String getAccountId(Context ctx, String accountName){
          ......
        return getToken(ctx, accountName, "^^_account_id_^^", new Bundle());
    }
```
AccountTrackerService: https://chromium.googlesource.com/chromium/src/+/bc808d5d314a6dca042f94c018ca6b520fc8e787/chrome/android/java/src/org/chromium/chrome/browser/signin/AccountTrackerService.java

AccountIdProvider: https://chromium.googlesource.com/chromium/src/+/d1fc9f028a155700ba07e34a9104d59d064e1763/chrome/android/java/src/org/chromium/chrome/browser/signin/AccountIdProvider.java



Also the AccountManager contains this code:
```
            if (res.auth != null) {
                Log.d(TAG, "getToken: " + res);
                result.putString(KEY_AUTHTOKEN, res.auth);
                Bundle details = new Bundle();
                details.putParcelable("TokenData", new TokenData(res.auth, res.expiry, scope.startsWith("oauth2:"), getScopes(scope)));
                result.putBundle("tokenDetails", details);
                result.putString(KEY_ERROR, "OK");
```

Which Sets `Error` to `"OK"`, I dont know if thats correct or not.

So it finally seems that `^^_account_id_^^` is some internal undocumented scope for Account-ID-fetching. I dont really know if this isn't an information leak considering the App already seems to know the Account mail address.

**Comment by mar-v-in** [comment]
Remove and recreate the account. This was fixed in 8df631cd950e83ed6881b939d39ac60601be67e2 but requires account recreation.

**Comment by lambdaupb** [comment]
My oversight!

I updated to the latest version and redid my account and everything is working nicely now.

Thanks :)
